src/Sparrow/Json/BlittableJsonDocumentBuilder.cs

```diff
@@ -294,6 +309,127 @@ private bool ReadInternal<TWriteStrategy>() where TWriteStrategy : IWriteStrateg
+                            }
+
+                            currentState.State = ContinuationState.CompleteArray;
+                            goto case ContinuationState.CompleteArray;
+                        }
+
+                        bool processed = false;
+
+                        // We try to read integers and doubles, if the value is a decimal (which can happen)
+                        // then we bail out because we cannot handle it as a vector (as least for now). 
+                        JsonParserToken current = _state.CurrentTokenType;
+
+                        switch (current)
+                        {
+                            case JsonParserToken.Integer:
+                                _state.AddBuffered(_state.Long);
+                                processed = true;
+                                break;
+                            case JsonParserToken.Float:
+                                var numberString = new ReadOnlySpan<byte>(_state.StringBuffer, _state.StringSize);
+                                if (Utf8Parser.TryParse(numberString, out decimal value, out int bytesConsumed) == false)
+                                {
+                                    //We suspect the underlying value might be a double with large values or exponents, so let's verify it again before moving to classic array.
+                                    if (Utf8Parser.TryParse(numberString, out double dValue, out bytesConsumed) == false)
+                                    {
+                                        break;
+                                    }
+                                    
+                                    _state.AddBuffered(dValue);
+                                    processed = true;
+                                    break;
+                                }
+
+                                Debug.Assert(bytesConsumed == _state.StringSize);
+#if NET7_0_OR_GREATER
+                                // This will only be executed on the server, so no problem on this regard as vector
+                                // representations are not supported on older systems.
+                                if (decimal.IsInteger(value) && decimal.Abs(value) < long.MaxValue)
+                                {
+                                    _state.AddBuffered(decimal.ToInt64(value));
+                                }
+                                else
+                                {
+                                    if ((_mode & UsageMode.ValidateDouble) == UsageMode.ValidateDouble)
+                                        _reader.ValidateFloat();
+
+                                    _state.AddBuffered(decimal.ToDouble(value));
+                                }
+#else
+                                throw new NotSupportedException("Vector representations are only supported for .Net versions greater than 6.0");
+#endif
+                                processed = true;
```

Comment:
Please check `Unreachable code detected` warning below

Reply:
Handled. It was an issue when `.NET` version is below v7.0.